pkg/autodiscovery/procdiscovery/discover.go

```diff
@@ -0,0 +1,110 @@
+package procdiscovery
+
+import (
+	"encoding/json"
+	"expvar"
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/procmatch"
+)
+
+// DiscoverIntegrations retrieves processes running on the host and tries to find possible integrations
+func DiscoverIntegrations(discoverOnly bool) (DiscoveredIntegrations, error) {
```

Comment:
I'm not a huge fan of interfaces where you have a bool that changes the behavior and return like this. Especially because of as a reader of the API it's very hard to discern was `DiscoveryIntegrations(true)` will do vs `DiscoverIntegrations(false)` Should these just be split up into two functions?

It feels like this interface should return _just_ a `map[string][]IntegrationProcess` and getting running checks should be different.